daemon/api.go

```diff
@@ -305,11 +306,23 @@ func loginUser(c *Command, r *http.Request, user *auth.UserState) Response {
 	overlord := c.d.overlord
 	state := overlord.State()
 	state.Lock()
-	_, err = auth.NewUser(state, loginData.Username, loginData.Email, macaroon, []string{discharge})
-	state.Unlock()
-	if err != nil {
-		return InternalError("cannot persist authentication details: %v", err)
+	// TODO Look at the error and fail if there's an attempt to authenticate with invalid data.
+	localUser, _ := UserFromRequest(state, r)
```

Comment:
From testing, it looks like loginUser is not going through the UserFromRequest call? Reviewing this to confirm.